### DISCUSSION TITLE
Is there a way to trigger the GitHub Git: Clone action on initial startup?

### DISCUSSION THREAD

**Discussion by binduwavell** [coder/code-server#3549] [Q&A] [answered]
I'm using the docker container. The startup script does a git clone... However, I have a private repo so that doesn't work unless I create some .ssh certs and load them into the container... Which is OK but them I'm storying those creds in my registry... Since VSCode/code-server has the git: clone feature, I was wondering if there is some sort of startup script where I could cause code-server to trigger this action, ideally with a preconfigured github project and local path?

**Comment by jsjoeio** [comment, marked as answer]
🤔 I'm not sure if that's possible. @bpmct or @code-asher might know though

**Reply by bpmct** [reply]
This can be done with tasks and the [AutoLaunch](https://marketplace.visualstudio.com/items?itemName=philfontaine.autolaunch) extension, someone just pointed this out to me the other day:

Task:
```
    {
      "label": "clone",
      "type": "shell",
      "command": "cd /home/coder/project && git clone $GIT_REPO .",
      "group": "build",
      "presentation": {
        "reveal": "always"
      },
      "problemMatcher": [],
      "options": {
        "statusbar": {
          "label": "$(clone) Clone"
        }
      }
    }
```
